subsys/senss/include/senss_sensor.h

```diff
@@ -0,0 +1,605 @@
+ * sensor instances
+ *
+ * @return number of reporters found, 0 returned if not found.
+ */
+int senss_sensor_get_reporters(
+		const struct device *dev, int type,
+		const senss_sensor_handle_t *reporter_handles, int max_handles);
+
+/**
+ * @brief Get reporters count of a given sensor instance by sensor type.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param type The sensor type for checking, \ref SENSS_SENSOR_TYPE_ALL
+ *
+ * @return Count of reporters by \p type, 0 returned if no reporters by \p type.
+ */
+int senss_sensor_get_reporters_count(
+		const struct device *dev, int type);
+
+/**
+ * @brief Get this sensor's state
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param state Returned sensor state value
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_sensor_get_state(
+		const struct device *dev,
+		enum senss_sensor_state *state);
+
+
+/**
+ * @}
+ */
+
+/**
+ * @brief Sensor Callbacks
+ * @addtogroup senss_sensor_callbacks
+ * \{
+ */
+
+/**
+ * @brief Sensor initialize callback for virtual sensor.
+ *
+ * Sensor can initialize it's runtime context in this callback.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param info The sensor instance's constant information.
+ *
+ * @param reporter_handles The reporters handles for this sensor.
+ *
+ * @param reporter_count The number of reporters.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+typedef int (*senss_sensor_virtual_init_t)(
+		const struct device *dev, const struct senss_sensor_info *info,
+		const senss_sensor_handle_t *reporter_handles, int reporters_count);
+
+/**
+ * @brief Sensor initialize callback for physical sensor.
+ *
+ * Sensor can initialize it's runtime context in this callback.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param info The sensor instance's constant information.
+ *
+ * @param underlying_dev The underlying sensor device from kernel space.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+typedef int (*senss_sensor_physical_init_t)(
+		const struct device *dev, const struct senss_sensor_info *info,
+		const struct device *underlying_dev);
+
+/**
+ * @brief Sensor's de-initialize callback.
+ *
+ * Sensor can release it's runtime context in this callback.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+typedef int (*senss_sensor_deinit_t)(
+		const struct device *dev);
+
+/**
+ * @brief Sensor reset callback.
+ *
+ * Sensor can reset its runtime context in this callback to default values without resources
+ * release and re-allocation.
+ *
+ * Its very useful for a virtual sensor to quickly reset its runtime context to a default state.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+typedef int (*senss_sensor_reset_t)(
```

Comment:
Is this short hand for `init` -> `deinit`? What's the expected difference?

Reply:
`reset` is typically for reset a virtual sensor's computing context, just reset some context variables to default value, etc,  so, it's very fast. 

`init` -> `deinit` -> `init` can do such thing, but will more slow, such as in `init`, the virtual sensor usually need to set some configurations to it's reporters, report interval, sensitivity etc.